src/Sylius/Bundle/CoreBundle/EventListener/Workflow/OrderCheckout/SkipPaymentListener.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\EventListener\Workflow\OrderCheckout;
+
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Order\StateResolver\StateResolverInterface;
+use Symfony\Component\Workflow\Event\CompletedEvent;
+use Webmozart\Assert\Assert;
+
+final class SkipPaymentListener
```

Comment:
I would be for:
```suggestion
final class ResolveOrderCheckoutStateListener
```
and linking it to both events, skipping payment and shipment